game/savemanager.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import logging
+import pickle
+import shutil
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import TYPE_CHECKING
+from zipfile import ZIP_LZMA, ZipFile
+
+from qt_ui import liberation_install
+from .profiling import logged_duration
+
+if TYPE_CHECKING:
+    from .game import Game
+
+
+class SaveGameBundle:
+    """The bundle of saved game assets.
+
+    A save game bundle includes the pickled game object (as well as some backups of
+    other game states, like the turn start and previous turn) and the state.json.
+    """
+
+    def __init__(self, bundle_path: Path) -> None:
+        self.bundle_path = bundle_path
+
+    def save_latest(self, game: Game) -> None:
+        with logged_duration("Saving game"):
+            self._update_bundle_member(game, self.latest_name)
+
+    def save_last_turn(self, game: Game) -> None:
```

Comment:
This is the state before applying the save.json. There's that, a start of turn (in case you make a mistake and want to reload to the start of the turn), and the main save for whenever you push the save button. You can't actually _do_ anything with most of that yet since there's no UI for loading the other states, but there will be.

Reply:
Added docstrings explaining the purpose of each.